### DISCUSSION TITLE
starting  ipsec  without having to run swanctl --load-all

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [strongswan/strongswan#2128] [Q&A]
currently I am testing with 

```
ipsec start
swanctl --load-all
```

I need to finish in my container entrypoint with this (or anything that runs in foreground)
```
/usr/sbin/ipsec start --nofork
```

How should I start strongswan?

**Comment by tobiasbrunner** [comment]
You don't need `ipsec` (it's only necessary for the old config format), just start charon directly. And you can load the config with a start script in strongswan.conf (`charon.start-scripts`).

**Comment by f1-outsourcing** [comment]
I am also getting such permission errors. Is this not a default setup they are using in alpine linux.

```
opening directory '/etc/swanctl/private' failed: Permission denied
opening directory '/etc/swanctl/rsa' failed: Permission denied
opening directory '/etc/swanctl/ecdsa' failed: Permission denied
opening directory '/etc/swanctl/bliss' failed: Permission denied
opening directory '/etc/swanctl/pkcs8' failed: Permission denied
opening directory '/etc/swanctl/pkcs12' failed: Permission denied

```
